File: imagesAndReprToTxtDIRTY.py
#!/usr/bin/python
#coding: utf-8

from __future__ import division
from torch.utils.serialization import load_lua

import numpy as np
import torch
from torchvision import transforms


import os

from PIL import Image

imagesFolder = 'simpleData3D'

modelString = 'dummyModelForDavid.t7'
model = load_lua(modelString)

reformatPipe = transforms.Compose([
    transforms.Scale(200),
    transforms.CenterCrop((200,200)),
    transforms.ToTensor()])

for seq in os.listdir(imagesFolder):
    strSeq = imagesFolder+'/'+seq+'/recorded_cameras_head_camera_2_image_compressed/'

    for imagePath in os.listdir(strSeq):
        if imagePath == 'time.txt':
            continue
            
        img = Image.open(strSeq+imagePath)
        img = reformatPipe(img)
        #NEED TO ADD NORMALIZATION AT A MOMENT

        representationOut = model.forward(img)
        print "representationOut",representationOut 
        
        

File: imagesAndReprToTxtDIRTY.lua
require 'lfs'
require 'torch'
require 'image'
require 'nn'
require 'nngraph'
require 'cunn'

require 'const'

local imagesFolder = 'simpleData3D'
local modelString = 'Log/model2017_137__9_55_55.t7'

local model = torch.load(modelString):cuda()
outStr = ''

for seqStr in lfs.dir(imagesFolder) do
   if string.find(seqStr,'record') then
      print("Sequence : ",seqStr)
      local imagesPath = imagesFolder..'/'..seqStr..'/recorded_cameras_head_camera_2_image_compressed'
      for imageStr in lfs.dir(imagesPath) do
         if string.find(imageStr,'jpg') then
            local fullImagesPath = imagesPath..'/'..imageStr
            local reprStr = ''
            local img=image.load(fullImagesPath,3,'float')

            img=image.scale(img,'200x200'):float():reshape(1,3,200,200):cuda()
            
            repr = model:forward(img)

            for i=1,repr:size(1) do
               reprStr = reprStr..repr[i]..' '
            end

            outStr = outStr..fullImagesPath..' '..reprStr..'\n'
            
         end
      end
   end
end        
        
file = io.open('saveImagesAndRepr.txt', 'w')
file:write(outStr)
